Link fixes + SessionSigs Updates

awisniew207 opened the pull request:
Changed all v3/v5 links I could find to v6, tested their functionality as well
Some bug fixes
Changed API Docs href to v6
Started making changes to the SessionSigs explanation

spacesailor24 commented on docs/sdk/migrations/6.0.0.md:
```diff
@@ -289,7 +289,7 @@ In the `signSessionKey` parameters, we've added three extra optional arguments f
 For simplicity, we introduced `getPkpSessionSigs`, which uses `signSessionKey` under the hood.
 
-Here's the (AuthCalbackParams)[https://github.com/LIT-Protocol/js-sdk/blob/635ec8da948103f3ce14271ccfc00393bc9e02a2/packages/types/src/lib/interfaces.ts#L45] interface for the `props`
+Here's the [AuthCalbackParams](https://github.com/LIT-Protocol/js-sdk/blob/635ec8da948103f3ce14271ccfc00393bc9e02a2/packages/types/src/lib/interfaces.ts#L45) interface for the `props`
```
```suggestion
Here's the [AuthCallbackParams](https://github.com/LIT-Protocol/js-sdk/blob/635ec8da948103f3ce14271ccfc00393bc9e02a2/packages/types/src/lib/interfaces.ts#L45) interface for the `props`
```

awisniew207 replied:
I noticed one error but not the other haha

spacesailor24 commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -12,11 +12,12 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
 
-We refer to a session signature obtained from the user via session keys as a `SessionSig`.
+Session keys are a unique string of data generated during a specific online session between two or more entities. In our case, the keypair is a ed25519 keypair. It serves as a protective layer for identity verification by incorporating unique session-specific details such as date, time, and other contextual factors to generate a signature. A signature, known as a `SessionSig`, is obtained from th ...
 
-`SessionSigs` are produced by a ed25519 keypair that is generated randomly on the browser and stored in local storage. The first step to producing `SessionSigs` is to first obtain an `AuthSig` through an authentication method like Google OAuth (example [here](https://github.com/LIT-Protocol/oauth-pkp-signup-example/blob/main/src/App.tsx#L398)). By specifying the session keypair's public key in th ...
+The Lit SDK uses the session keypair to sign all requests to the Lit Nodes. Each node in the Lit Network receives a unique signature for each request and can verify that the user owns the wallet address that signed the capability. The user’s `AuthSig` is sent along with the `SessionSig` as an attached "capability" to introduce capability to the `SessionSig`.
+
+Attaching capability is done by specifying the session keypair's public key as the `address` field of the `AuthSig` and the `capabilities` field. The `capabilities` field is an array of one or more signatures which authorize the `AuthSig` address (also our session keypair public key) to utilize the resources specified in the capabilities of the SIWE messages. These signatures contain the `AuthSig ...
```
@awisniew207 how do you feel about rewriting the entire intro to:

In order to communicate with a Lit network, you must first authenticate yourself (i.e. prove who you are) with the network - this is done via Session Signatures.

Session Signatures are made using Session Keys which are generated on your behalf when you start a session by making a request to a Lit network using the Lit SDK. These Session Keys are a unique `ed25519` keypair generated by the Lit SDK locally, and are used to sign all requests to the Lit network for the current session.

When generating the Session Signatures, an Authentication Signature (Auth Sig) is used to prove who are to the Lit nodes, so that they can check if you're authorized to perform the action you're requesting to be done by the Lit network e.g. decrypting some data, signing a transaction with a PKP, transferring ownership of a PKP.

The Auth Sig is a [ERC-5573](https://eips.ethereum.org/EIPS/eip-5573) Sign-In with Ethereum Capabilities message that specifies a set of Lit resources (e.g. PKPs, Lit Actions) and specific [Lit Abilities](https://v6-api-doc-lit-js-sdk.vercel.app/enums/types_src.LitAbility.html) that the Session Keys are being granted usage of e.g. Signing transactions with a specific PKP that is owned by the address that produced the Auth Sig, execution of a specified Lit Action.

After making your request to the Lit network, each Lit node derives an identity from the Auth Sig and uses it to check if that identity owns/is authorized to use the Lit resource(s) and corresponding abilities your request is attempting to use.

awisniew207 commented:
Introduced working examples for getSessionSigs, getPkpSessionSigs, getLitActionSessionSigs. Added a little explanation.
Rewrote the introduction.
Just need review and feedback.

a1ttech commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
```diff
@@ -6,35 +6,59 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
+These functions use the session keypair generated by the `LitNodeClient` to invoke a callback function that signs the generated session key, creating an `AuthSig` that is scoped to specific capabilities.
 
-You can generate a `SessionSig` with the help of `capacityDelegationAuthSig` object in the following way:
+In order to generate `SessionSigs` on both the `Habanero` and `Manzano` networks, you will need to include `capabilityAuthSigs`. This is because transactions on these networks require capacity credits. `capabilityAuthSigs` are **not** required on the `datil-dev` network.
```
we might also want to reference the 'upcoming' datil-test and datil-prod networks as well here

a1ttech commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
````diff
@@ -73,39 +97,194 @@ const sessionSigsFunction = async() => {
- ```
-:::
+## getPkpSessionSigs()
+This is an implementation of `getPkpSessionSigs()`. Noticably different from `getSessionSigs()`, it lacks the `authNeededCallback` parameter, instead creating the callback and signing the session key under the hood with the `signSessionKey()` function. [API Docs](https://v6-api-doc-lit-js-sdk.vercel.app/classes/lit_node_client_src.LitNodeClientNodeJs.html#getLitActionSessionSigs)
````
can we specify why someone would want to use one of these functions over the other? Like hint at recommended usage ('we recommend using this function if you are...') perhaps in the overview at the beginning of this section?

awisniew207 replied:
@a1ttech Changed this, would appreciate confirmation on the accuracy of my comments 😅:

We recommend using `getSessionSigs()` to specify the abilities of your current session on the Lit Network, using only an `ethers.Wallet` and the `LitNodeClient`. This method is arguably the simplest, enabling your `ethers.Wallet` to perform the specified abilities on the Lit Network. `getPkpSessionSigs()` can be used if your implementation of the Lit SDK involves a PKP and you wish to designate resources to it. Similarly, `getLitActionSessionSigs()` can designate specific abilities and resources to Lit Actions, potentially allowing them to execute code on the Lit Network.

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
```diff
@@ -3,112 +3,256 @@ sidebar_position: 2
+This guide covers the `getSessionSigs` function from the Lit SDK. For an overview of what session signatures are and how they are to be used, please go [here](./intro).
 
-In order to generate `SessionSigs`, you need a `capacityDelegationAuthSig` in signatures. This is because capacity credits are now required on both `Habanero` and `Manzano` networks. You can buy or [mint](../../capacity-credits.md#minting-capacity-credits) capacity credits centrally and [delegate](../../capacity-credits.md#delegating-access-to-your-capacity-credits-nft) them to your users. 
+The function uses the session keypair generated by the `LitNodeClient` to invoke a callback function that signs the generated session key, creating an `AuthSig` that is scoped to specific capabilities.
```
```suggestion
The function uses the session keypair generated by the `LitNodeClient` to invoke a callback function that signs the generated session key, creating an `AuthSig` that is scoped to specific Lit capabilities.
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
```diff
@@ -3,112 +3,256 @@ sidebar_position: 2
+The function uses the session keypair generated by the `LitNodeClient` to invoke a callback function that signs the generated session key, creating an `AuthSig` that is scoped to specific capabilities.
 
-You can generate a `SessionSig` with the help of `capacityDelegationAuthSig` object in the following way:
+Using the `getSessionSigs` function, you can specify the abilities of your current session on the Lit network. This function is the simplest way to get session signatures, at the minimum only requiring an `ethers.Wallet` and the `LitNodeClient`. It will enable your `ethers.Wallet` to perform abilities to the resources you specify in the `AuthSig`.
```
```suggestion
Using the `getSessionSigs` function, you can specify the Lit capabilities of your current session on the Lit network. This function is the simplest way to get session signatures, at the minimum only requiring an `ethers.Wallet` and the `LitNodeClient`. It will enable your `ethers.Wallet` to perform abilities to the resources you specify in the `AuthSig`.
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
````diff
@@ -3,112 +3,256 @@ sidebar_position: 2
-  * When the getSessionSigs function is called, it will generate a session key
-  * and sign it using a callback function. The authNeededCallback parameter
-  * in this function is optional. If you don't pass this callback,
-  * then the user will be prompted to authenticate with their wallet.
-  */
-  const authNeededCallback = async ({
+```ts
+interface CommonGetSessionSigsProps {
+  pkpPublicKey?: string;
+  expiration?: any;
+  chain?: Chain;
+  resourceAbilityRequests: LitResourceAbilityRequest[];
+  sessionCapabilityObject?: ISessionCapabilityObject;
+  switchChain?: boolean;
+  sessionKey?: SessionKeyPair;
+  capabilityAuthSigs?: AuthSig[];
+}
+
+interface GetSessionSigsProps
+  extends CommonGetSessionSigsProps,
+    LitActionSdkParams {
+  authNeededCallback: AuthCallback;
+}
+
+getSessionSigs = async (
+    params: GetSessionSigsProps
+  ): Promise<SessionSigsMap>
+
+interface AuthSig {
+  sig: any;
+  derivedVia: string;
+  signedMessage: string;
+  address: string;
+}
+
+type SessionSigsMap = Record<string, AuthSig>;
+```
````
From Daryl's feedback, we shouldn't include links to the source code or have the source code included in the docs because this is hard to maintain whenever there's a change to the interface within the SDK. So, this section should be something like:

```
## `getSessionSigs` Interface

For details on this methods interface, please go [here](https://v6-api-doc-lit-js-sdk.vercel.app/classes/lit_node_client_src.LitNodeClientNodeJs.html#getSessionSigs).
```

Also open to other ideas of how this should look like if you have them

spacesailor24 replied:
And this would imply we remove the `Parameters` and `Return Value` sections, as this info should be in the SDK API docs. For this I think we should review the SDK API docs and if the information isn't better than what you have here, we should make a PR into the SDK to improve the docs

awisniew207 replied:
I guess if Daryl doesn't recommend any source code (snippits or links) in the docs then we can only reference the API docs. I agree with just linking the API docs then.

I based my `Parameters` and `Return Value` sections off of the `Wrapped Keys` guides, with most of the information being directly from the SDK (so via the API docs). I think explaining parameters/returns in the docs is worthwhile even if the information is in the API docs as well, looks better and makes the concepts more understandable as well IMO. Could we get a third opinion on this?

spacesailor24 replied:
> I think explaining parameters/returns in the docs is worthwhile even if the information is in the API docs as well, looks better and makes the concepts more understandable as well IMO

I agree, but the problem Daryl raised is we now have to manage the information in both the docs and the SDK API docs, instead of just directing users to the API docs which will (in theory) always be kept up to date since the docs live next to the code

If the SDK API docs aren't covering what each parameter/return value is sufficiently, then we should improve those docs, not write them in two places

awisniew207 replied:
Alright, I removed the sections from the PR. I don't think the SDK API Docs are very good, so I'll make a PR with better explanations

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
```diff
@@ -3,112 +3,256 @@ sidebar_position: 2
+
+The full code implementation can be found [here](https://github.com/LIT-Protocol/developer-guides-code/tree/master/session-signatures/getSessionSigs). 
+
+### Installing the Required Dependencies
```
We should add a note to this section that the `node-localstorage` dependency is only required when executing outside of a browser environment since the SDK will use the native browser storage

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
```diff
@@ -3,112 +3,256 @@ sidebar_position: 2
+import * as ethers from "ethers";
+
+const ethersSigner = new ethers.Wallet(
+  ETHEREUM_PRIVATE_KEY,
```
```suggestion
  process.env.ETHEREUM_PRIVATE_KEY,
```

We should always prefix ENVs with `process.env.`

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
````diff
@@ -3,112 +3,256 @@ sidebar_position: 2
+```
+
+### Instantiating a `LitNodeClient`
+Here we are instantiating an instance of `LitNodeClient` and connecting it to the `datil-dev` Lit network. To simulate a Node.js environment, we use `LocalStorage` from the `node-localstorage` module to store our session keys and metadata.
````
```suggestion
Here we are instantiating an instance of `LitNodeClient` and connecting it to the `datil-dev` Lit network. Because we're executing within a Node.js environment, we use `LocalStorage` from the `node-localstorage` module to store our session keys and metadata.
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
````diff
@@ -3,112 +3,256 @@ sidebar_position: 2
+```
+
+### Generating Session Signatures
+The session signatures generated in this example provide unrestricted access to all resources. However, in real-world applications, it's more common and secure to limit access. Typically, you would use the specific hash of the encrypted access control condition resource key, granting access only to particular resources rather than everything. To this resource, we then give it the ability to decry ...
````
```suggestion
In this example, we're granting the capability to request to decrypt any data that we may be authorized to decrypt (i.e. we satisfy the Access Control Conditions the data was encrypted with). We could, however, specify the [LitAccessControlConditionResource](https://v6-api-doc-lit-js-sdk.vercel.app/classes/auth_helpers_src.LitAccessControlConditionResource.html) for specific encrypted data we're permitting the decryption capability for. In real-world applications, it's more common and secure to limit access to specific Lit resources instead of specifying the wildcard (`"*"`) identifier.

To get the Lit resource identifier for other resources, you can use the other methods included in [@lit-protocol/auth-helpers](https://v6-api-doc-lit-js-sdk.vercel.app/modules/auth_helpers_src.html) package.
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
```diff
@@ -3,112 +3,256 @@ sidebar_position: 2
+      expiration,
       resources: resourceAbilityRequests,
-      walletAddress: wallet.address,
+      walletAddress: await ethersSigner.getAddress(),
```
```suggestion
      walletAddress: wallet.address,
```

This should work

awisniew207 replied:
Wallet isn't defined in the code example, ethersSigner.address seems to be alright

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
```diff
@@ -3,112 +3,256 @@ sidebar_position: 2
-The `getSessionSigs()` function will try to create a session key for you and store it in local storage. You can also generate the session key yourself using `generateSessionKeyPair()` function and store it however you like. You can then pass the generated session key to `getSessionSigs()` as the `sessionKey` param.
-
-In the example above, we construct a SIWE message manually, but you can use the `checkAndSignAuthMessage` when in a browser context.
+**Note:** The nonce should be the latest Ethereum blockhash returned by the nodes during the handshake.
```
```suggestion
:::note
The nonce should be the latest Ethereum blockhash returned by the nodes during the handshake.
:::
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
```diff
@@ -3,112 +3,256 @@ sidebar_position: 2
 ## Resources You Can Request
 
-You can pass an array of "resource ability requests" to the `getSessionSigs()` function, which will be presented to the user in the SIWE message - read more [here](resources-and-abilities) about Lit resources and abilities. The resources and abilities requested by the session key must be narrower or equal to the capabilities granted to it per the session capability object specified in the inner ` ...
+You can pass an array of `resourceAbilityRequests` to the above functions, which will be presented to the user in the SIWE message - read more [here](resources-and-abilities) about Lit resources and abilities. The resources and abilities requested by the session key must be narrower or equal to the capabilities granted to it per the session capability object specified in the inner `AuthSig`. 
```
```suggestion
You can pass an array of `resourceAbilityRequests` to the `getSessionSigs` function, which will be presented to the user in the SIWE message - read more [here](resources-and-abilities) about Lit resources and abilities. The resources and abilities requested by the session key must be narrower or equal to the capabilities granted to it per the session capability object specified in the inner `AuthSig`. 
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
```diff
@@ -3,112 +3,256 @@ sidebar_position: 2
-In the example above, we construct a SIWE message manually, but you can use the `checkAndSignAuthMessage` when in a browser context.
+**Note:** The nonce should be the latest Ethereum blockhash returned by the nodes during the handshake.
 
 ## Resources You Can Request
```
I feel like this section can be removed, and we should cover it in the `Session Signatures` overview page, or in the specific page `Lit Resources and Abilities`

I'd actually like to redo the `Lit Resources and Abilities` and `Capability Objects` pages, but that's out of scope for this PR

awisniew207 replied:
I'll add this to the overview

awisniew207 replied:
Will say this in the overview. I do think the `Lit Resources and Abilities` page could be redone, maybe I get to it after wrapping up session sigs?

spacesailor24 replied:
I've created [a ticket](https://linear.app/litprotocol/issue/DREL-69/refactor-the-lit-resources-and-abilities-docs-page) for it, but we should discuss priority in the next stand up

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
```diff
@@ -3,112 +3,256 @@ sidebar_position: 2
-When session capability objects are omitted from the `getSessionSigs()` function call, the SDK will generate a session capability object with **wildcard permissions against all of the resources in that category by default**, i.e. ability to perform operations against all access control conditions. Read more [here](capability-objects) about how to create custom session capability objects.
+When session capability objects are omitted from functions generating session signatures, the SDK will generate a session capability object with **wildcard permissions against all of the resources in that category by default**, i.e. ability to perform operations against all access control conditions. Read more [here](capability-objects) about how to create custom session capability objects.
 
 ## Clearing Local Storage
```
Should add an explanation of why you'd want to do this. Although I'm not sure why you'd want to outside of debugging

awisniew207 replied:
Will say this in the overview

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
```diff
@@ -3,112 +3,256 @@ sidebar_position: 2
+## Summary
+The full code implementation can be found [here](https://github.com/LIT-Protocol/developer-guides-code/tree/master/session-signatures/getSessionSigs). 
+
+After executing the example implementation above, you will have generated and stored your session public and secret keys, and the `AuthSig` signed for the session signatures. 
```
```suggestion
After executing the example implementation above, you will have generated and stored session signatures that allow you to request to decrypt data that you satisfy the Access Control Conditions for.

You can use these session signatures in a decryption request [like so](../../../access-control/quick-start#decryption).
```

awisniew207 replied:
What about:

After executing the example implementation above, you will have generated and stored session signatures that allow requests to decrypt data that match your Access Control Conditions.

spacesailor24 replied:
```
After executing the example implementation above, you will have generated and stored session signatures that allow requests to decrypt data that you satisfy the Access Control Conditions for.
```

Were not matching the control conditions, they're a set of constraints that we're satisfying

a1ttech commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
 `SessionSigs` are only available on Ethereum and are heavily in development, so things may change. Be sure to use the latest version of the Lit JS SDK.
 
 :::
+Session signatures are how the Lit network authenticates your connection, verifies your identity, and confirms your capabilities. 
```
i think we could simplify this a bit. Perhaps someting along the lines of "Session signatures are used to authenticate with the Lit nodes and create a secure connection to the Lit network"

spacesailor24 replied:
I've been capitalizing Lit jargon since it's a specific thing in our ecosystem. My preference is to always capitalize Session Signatures

a1ttech commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
+Session signatures are how the Lit network authenticates your connection, verifies your identity, and confirms your capabilities. 
 
-We refer to a session signature obtained from the user via session keys as a `SessionSig`.
+Generating session signatures is necessary whenever you want to request [Lit Abilities](https://v6-api-doc-lit-js-sdk.vercel.app/enums/types_src.LitAbility.html) (e.g., signing transactions with a particular PKP, executing a specified Lit Action) for your Lit Resources (e.g. PKPs, Lit Actions).
```
> Generating session signatures is necessary whenever you want to request [Lit Abilities](https://v6-api-doc-lit-js-sdk.vercel.app/enums/types_src.LitAbility.html) (e.g., signing transactions with a particular PKP, executing a specified Lit Action) for your Lit Resources (e.g. PKPs, Lit Actions).

Consider "Generating a session signature is required whenever you want to request a specific [Lit ability] (e.g. signing a transaction) for a particular Lit resource (e.g. a PKP)"

spacesailor24 replied:
```suggestion
Generating session signatures is necessary whenever you want to request [Lit Abilities](https://v6-api-doc-lit-js-sdk.vercel.app/enums/types_src.LitAbility.html) (e.g. signing transactions with a particular PKP, executing a specified Lit Action) for your Lit Resources (e.g. PKPs, Lit Actions).
```

a1ttech commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
+Usage of the `datil` and `datil-test` networks require the use of [Lit Capacity Credits](../../../sdk/capacity-credits.md). Currently, Capacity Credits are paid for using the `tstLPX` token and don't require any real-world money. However, in the future you will need to pay real-world money for usage of Lit networks, and `datil` and `datil-test` are the Lit networks where this functionality is bei ...
 
-<br/>
+To implement payments correctly, include a [`capacityDelegationAuthSig`](https://developer.litprotocol.com/sdk/capacity-credits#createcapacitydelegationauthsig) within the `capabilityAuthSigs` array when generating session signatures.
```
we can link to Wyatt's new payment docs here when they're ready

a1ttech commented on docs/sdk/authentication/session-sigs/get-lit-action-session-sigs.md:
```diff
@@ -0,0 +1,168 @@
+
+Using the `getLitActionSessionSigs` function, you can specify the capabilities of your current session on the Lit network. 
+
+This function is very similar to [`getPkpSessionSigs`](./get-pkp-session-sigs.md). The `getPkpSessionSigs` function requires you to own a PKP and some form of authentication to prove your identity (e.g. a custom Lit Action, AuthMethod, or AuthSig).
```
it could be helpful to add a few more sentences here that clearly articulate the differences between these two functions and why one would use one over the other

awisniew207 replied:
I think this is the only major difference? Most of the `getLitActionSessionSig` properties are inherited from `getPkpSessionSigs`. Their generated session signatures are identical as well:

> This function is very similar to [`getPkpSessionSigs`](./get-pkp-session-sigs.md). The `getPkpSessionSigs` function requires you to own a PKP and some form of authentication to prove your identity (e.g. a custom Lit Action, AuthMethod, or AuthSig).
> 
> Alternatively, the `getLitActionSessionSigs` function requires the form of authentication to be a Lit Action. Other authentication methods can be included as well, but a Lit Action is required.

awisniew207 commented on docs/sdk/authentication/session-sigs/get-pkp-session-sigs.md:
```diff
@@ -0,0 +1,148 @@
+## Summary
+The full code implementation can be found [here](https://github.com/LIT-Protocol/developer-guides-code/tree/master/session-signatures/getPkpSessionSigs). 
+
+After executing the example implementation above, you will have generated session signatures that allow you to request decrypting data that you have satisfied the Access Control Conditions for.
```
Note: The last sentence of `getPkpSessionSigs` is wrong, it will be fixed in the next push

a1ttech commented on docs/sdk/authentication/session-sigs/get-lit-action-session-sigs.md:
````diff
@@ -0,0 +1,168 @@
+```
+
+### Generating Session Signatures
+In this example, we're granting the capability to request to decrypt any data that we may be authorized to decrypt (i.e. we satisfy the Access Control Conditions the data was encrypted with). We could, however, specify the [LitAccessControlConditionResource](https://v6-api-doc-lit-js-sdk.vercel.app/classes/auth_helpers_src.LitAccessControlConditionResource.html) for specific encrypted data we're  ...
````
i could be misunderstanding, but here you reference granting a capability to decrypt, but the code below is for PKP signing?

awisniew207 replied:
No you're right, is was decryption for a previous example and I had missed this change.

awisniew207 replied:
Changed to:

> In this example, we're enabling our session to use a PKP for signing and to execute Lit Actions.
> 
> The current code uses the wildcard (`*`) identifier for `LitPKPResource`, which grants signing abilities to **any** PKP. This should only be used for example implementations or debugging. A more secure implementation would instead use the PKP `tokenId` to grant a specific PKP signing abilities. 
> 
> The wildcard identifier is also used for `LitActionResource`. This allows the session to execute **any** Lit Action. A more secure implementation would instead have a specific IPFS CID.

a1ttech commented on docs/sdk/authentication/session-sigs/get-pkp-session-sigs.md:
````diff
@@ -0,0 +1,148 @@
+```
+
+### Generating Session Signatures
+In this example, we're granting the capability to request to decrypt any data that we may be authorized to decrypt (i.e. we satisfy the Access Control Conditions the data was encrypted with). We could, however, specify the [LitAccessControlConditionResource](https://v6-api-doc-lit-js-sdk.vercel.app/classes/auth_helpers_src.LitAccessControlConditionResource.html) for specific encrypted data we're  ...
````
same comment as above regarding decryption capabilities vs pkp signing

awisniew207 replied:
Similar to the Lit Actions change:

> In this example, we're enabling our session to use a PKP for signing.
> 
> The current code uses the wildcard (`*`) identifier for `LitPKPResource`, which grants signing abilities to **any** PKP. This should only be used for example implementations or debugging. A more secure implementation would instead use the PKP `tokenId` to grant signing abilities to a specific PKP.

spacesailor24 commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
+This authentication system enhances the security of the Lit network. For detailed explanations of this setup, please refer to our [Security Considerations](../security.md) page.
 
-## Format of `SessionSigs`
+## SessionSigs Generation Diagram
+![Session Signatures Diagram](../../../../static/img//SessionSigs.png)
```
Not sure what the fix is here, but wanted to call it out. When viewing on dark mode the arrows in the diagram are lost:

<img width="987" alt="image" src="https://github.com/user-attachments/assets/9d493068-015a-422d-9dff-eafaa1b39801">

spacesailor24 replied:
Oh we should export the diagram with a white background instead of transparent

spacesailor24 commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
 `SessionSigs` are only available on Ethereum and are heavily in development, so things may change. Be sure to use the latest version of the Lit JS SDK.
```
> `SessionSigs` are only available on Ethereum

I'm not exactly sure what this is trying to say, and I think it would confuse users since it sounds like you can only use Session Signatures if you're using Ethereum with Lit

awisniew207 replied:
Removed this. Was included only because the current docs have it

spacesailor24 commented:
We should remove the markdown ticks `` from the page titles and it doesn't render correctly on the sidebar:
<img width="261" alt="image" src="https://github.com/user-attachments/assets/a79a176b-c3de-42e8-b9f5-9b1621842723">

spacesailor24 commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
+Generating session signatures is necessary whenever you want to request [Lit Abilities](https://v6-api-doc-lit-js-sdk.vercel.app/enums/types_src.LitAbility.html) (e.g. signing transactions with a particular PKP, executing a specified Lit Action) for your Lit Resources (e.g. PKPs, Lit Actions).
 
-`SessionSigs` are produced by a ed25519 keypair that is generated randomly on the browser and stored in local storage. The first step to producing `SessionSigs` is to first obtain an `AuthSig` through an authentication method like Google OAuth (example [here](https://github.com/LIT-Protocol/oauth-pkp-signup-example/blob/main/src/App.tsx#L398)). By specifying the session keypair's public key in th ...
+Session signatures are created with session keys, which are generated when you initiate a session through a request to a Lit network using the Lit SDK. Session keys are unique [`ed25519`](https://ed25519.cr.yp.to/) keypairs generated locally by the Lit SDK and are used to sign all requests to the Lit Network during the current session. 
```
Consider something like:

```suggestion
Session signatures are created using session keys, which are generated for you when you initiate a connection with the Lit network via the Lit SDK. These session keys are unique [`ed25519`](https://ed25519.cr.yp.to/) keypairs generated locally by the Lit SDK. They are used to sign all requests to the Lit Network during the current session.
```

spacesailor24 replied:
Also are we sure it's `ed25519` keys and not `Secp256k1`?

awisniew207 replied:
Got this from the current docs:
```sig is the signature produced by the ed25519 keypair signing the signedMessage payload```

spacesailor24 commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
+Session signatures are created with session keys, which are generated when you initiate a session through a request to a Lit network using the Lit SDK. Session keys are unique [`ed25519`](https://ed25519.cr.yp.to/) keypairs generated locally by the Lit SDK and are used to sign all requests to the Lit Network during the current session. 
 
-The session keypair is used to sign all requests to the Lit Nodes, and the user's `AuthSig` is sent along with the request, attached as a "capability" to the session signature. Each node in the Lit Network receives a unique signature for each request, and can verify that the user owns the wallet address that signed the capability.
+While session signatures facilitate ongoing communication during a session, an `AuthSig` (Authentication Signature) is used to verify your identity and authorization to the Lit Nodes.
```
```suggestion
While session keys and their signatures facilitate ongoing communication during a session, an `AuthSig` (Authentication Signature) is used to verify your identity and authorization to the Lit Nodes.
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
+An `AuthSig` is an [ERC-5573](https://eips.ethereum.org/EIPS/eip-5573) Sign-In with Ethereum Capabilities message that specifies the Lit Resources and Lit Abilities of the session.
 
-Session signatures work by having scoped capabilities be granted to session keys by an inner `AuthSig`. The capability object is a [SIWE ReCap](https://eips.ethereum.org/EIPS/eip-5573) object.
+The `AuthSig` allows Lit Nodes to verify your authorization for requested actions, such as decrypting data, signing transactions with a PKP, or transferring PKP ownership. This ensures that only authorized users can perform specific actions within the Lit Network.
```
```suggestion
The `AuthSig` allows Lit nodes to verify your authorization for requested actions, such as decrypting data, signing transactions with a PKP, or transferring PKP ownership. This ensures that only authorized users can perform specific actions within the Lit network.
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/intro.md:
There's inconsistency with capitalization. We should do `Lit network` and `Lit nodes`

awisniew207 replied:
I changed this for this PR, I can make a PR later to change it everywhere in the docs if needed

spacesailor24 commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
 
-## Capability Objects
+An `AuthSig` is an [ERC-5573](https://eips.ethereum.org/EIPS/eip-5573) Sign-In with Ethereum Capabilities message that specifies the Lit Resources and Lit Abilities of the session.
 
-Session signatures work by having scoped capabilities be granted to session keys by an inner `AuthSig`. The capability object is a [SIWE ReCap](https://eips.ethereum.org/EIPS/eip-5573) object.
+The `AuthSig` allows Lit nodes to verify your authorization for requested actions, such as decrypting data, signing transactions with a PKP, or transferring PKP ownership. This ensures that only authorized users can perform specific actions within the Lit network.
+When you make a request, each Lit Node checks your `AuthSig` to confirm that your request aligns with the capabilities you previously defined. 
```
Consider:

An AuthSig is an [ERC-5573](https://eips.ethereum.org/EIPS/eip-5573) Sign-In with Ethereum Capabilities message. It defines the specific Lit Resources and Lit Abilities authorized for your session.

The AuthSig enables the Lit nodes to verify your authorization to perform various actions within the Lit network. These actions may include decrypting data, signing transactions with a PKP, or executing Lit Actions.

When you submit a request to the Lit network, each node checks the AuthSig contained within your Session Signatures to ensure that your request aligns with the capabilities granted for your session. This validation process guarantees that only authorized users can perform specific actions using the Lit Resources defined by the AuthSig, maintaining the security and integrity of the Lit network.

awisniew207 replied:
Blended them:

> An `AuthSig` is an [ERC-5573](https://eips.ethereum.org/EIPS/eip-5573) Sign-In with Ethereum Capabilities message. It specifies the authorized Lit Resources and Lit Abilities of the session.
> 
> The `AuthSig` allows Lit nodes to verify your authorization to perform requested actions, such as decrypting data, signing transactions with a PKP, or transferring PKP ownership. When you make a request, each Lit node checks your `AuthSig` to confirm that your request aligns with the capabilities you previously defined. This ensures that only authorized users can perform specific actions within the Lit network. This authentication system maintains the security and integrity of the Lit network.
> 
> For detailed explanations of this setup, please refer to our [Security Considerations](../security.md) page.

spacesailor24 replied:
> or transferring PKP ownership

This is done by transferring the underlying NFT which doesn't necessarily require a Session Sig. I want replace this with `or executing Lit Actions`

awisniew207 replied:
Changed!

spacesailor24 commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
+![Session Signatures Diagram](../../../../static/img//SessionSigs.png)
 
-Given the following example `AuthSig`:
+## Paying for Usage of the Lit Network
```
Is this section needed? The same info gets covered by the Delegation doc page in the Paying for Lit PR

awisniew207 replied:
I wouldn't say necessary, but maybe helpful to the page? How about I just keep the first two sentences and add a link to the Delegation doc page once its up?

spacesailor24 replied:
Ok, but I say something along the lines of:

You can facilitate payment for usage the Lit network within Session Signatures. You can read more about paying for usage [here], and paying using Session Signatures [here].

awisniew207 replied:
Will push this linking to the capacity credits pages and update with the proper links once #319 has been merged

spacesailor24 commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
-- `signedMessage` is the payload that was signed by the session keypair.
-- `address` is the session keypair public key.
-- `algo` is the signing algorithm used to generate the session signature.
+When running code to generate session signatures, you can store the session signatures and session keys using `LocalStorage` from the `node-localstorage` package.
```
```suggestion
When running code to generate session signatures in a Node.js environment, you can store the session signatures and session keys using `LocalStorage` from the `node-localstorage` package.
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/intro.md:
````diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
+```
 
-Here is what each field in `signedMessage` means:
+When running code within a browser, this import is not needed, as the session signatures and keys will be stored within the browser's local storage. However, when running this code in an environment such as Node.js where browser local storage is not available, the `LocalStorage` module can be used to provide file-based storage. 
 
-- `sessionKey` is the session keypair public key.
-- `resourceAbilityRequests` is a lit of abilities that the session key is requesting to perform against the specified Lit resources during authentication. Read more [here](resources-and-abilities) about Lit Resources and Abilities.
-- `capabilities` is an array of one or more AuthSigs.
-- `issuedAt` is the time the SessionSig was issued.
-- `expiration` is the time the SessionSig becomes invalid.
-- `nodeAddress` is the specific URL the SessionSig is meant for.
+```javascript
+litNodeClient = new LitNodeClient({
+    litNetwork: LitNetwork.DatilDev,
+    // This storageProvider object can be omitted if executing in a browser
+    storageProvider: {
+        provider: new LocalStorage("./lit_storage.db"),
+    },
+});
+```
````
Consider:

When generating Session Signatures, you can store the signatures and session keys for reuse until the expire. How you store them depends on your runtime environment:

#### Running in a Browser

Session data is automatically stored in the browser's local storage, and no additional configuration is needed.

#### Running in a Node.js (or Similar) Environment

A `storageProvider` needs to specified when creating an instance of the `LitNodeClient`. You decide how this implemented, but the following is an example of using `LocalStorage` from the `node-localstorage` package:

```ts
import { LocalStorage } from "node-localstorage";
import { LitNodeClient } from "@lit-protocol/lit-node-client";
import { LitNetwork } from "@lit-protocol/constants";

const litNodeClient = new LitNodeClient({
    litNetwork: LitNetwork.DatilDev,
    // This storageProvider object can be omitted if executing in a browser
    storageProvider: {
        provider: new LocalStorage("./lit_storage.db"),
    },
});
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
+When a `storageProvider` is added to the `LitNodeClient` instance, all functions for generating session signatures will try to create a session keypair for you and store it in the local storage. 
 
-#### Capabilities
+The session keypair can also be generated with the `generateSessionKeyPair()` function. Doing this enables you to pass the generated session keypair as the optional `sessionKey` parameter when generating session signatures.
```
This seems a little out of place for the storage section?

awisniew207 replied:
How about this below the local storage section?:

> #### Manually Generating Session Keys
> The session keypair can also be generated with the `generateSessionKeyPair()` function. Doing this enables you to pass the generated session keypair as the optional `sessionKey` parameter when generating Session Signatures.

spacesailor24 commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
+The session keypair can also be generated with the `generateSessionKeyPair()` function. Doing this enables you to pass the generated session keypair as the optional `sessionKey` parameter when generating session signatures.
 
-The `capabilities` field is an array of one or more signatures. These capabilities authorize this AuthSig address to utilize the resources specified in the capabilities SIWE messages. These signatures would have the address from the top level AuthSig in their URI field. For example, notice the following in the AuthSig above:
+If an instance of `LocalStorage` is not provided as the `storageProvider`, a new session keypair will be generated each time the code runs, instead of reusing a single set of keys.
```
```suggestion
If an instance of `LocalStorage` is not provided as the `storageProvider`, a new session keypair will be generated each time the code runs, instead of reusing previously generated signatures and keys.
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
+### Resources you can Request
 
-#### Node Address
+You can pass an array of `resourceAbilityRequests` to any of the functions that generate session signatures. These will be presented to the user in the SIWE message - read more [here](resources-and-abilities) about Lit resources and abilities. The resources and abilities requested by the session key must be narrower or equal to the capabilities granted to it per the session capability object spec ...
```
Consider:

When generating Session Signatures, you can pass an array of `resourceAbilityRequests` to the relevant functions. These requests define the scope of actions the session key can perform.

The requested resources and abilities must be either equal to or a subset of the capabilities specified in the session capability object within the `AuthSig`. A session key cannot have more privileges than those originally granted by the authentication signature.

awisniew207 replied:
Blended them:

> You can pass an array of `resourceAbilityRequests` to any of the functions that generate Session Signatures. These requests define the scope of actions the session key can perform. 
> 
> The requested resources and abilities must be either equal to or a subset of the capabilities specified in the session capability object within the `AuthSig`. A session key cannot have more privileges than those originally granted by the authentication signature.

spacesailor24 commented on docs/sdk/authentication/session-sigs/intro.md:
```diff
@@ -11,116 +11,63 @@ import FeedbackComponent from "@site/src/pages/feedback.md";
+You can pass an array of `resourceAbilityRequests` to any of the functions that generate session signatures. These will be presented to the user in the SIWE message - read more [here](resources-and-abilities) about Lit resources and abilities. The resources and abilities requested by the session key must be narrower or equal to the capabilities granted to it per the session capability object spec ...
 
-The `nodeAddress` will be different for each node, which means that, for a 30-node network, the SDK will generate 30 different `sig` and `signedMessage` parameters.
+When session capability objects are omitted from functions generating session signatures, the SDK will generate a session capability object with **wildcard permissions against all of the resources in that category by default**, i.e. ability to perform operations against all access control conditions. This should only be done when debugging, as allowing unspecified access control conditions is a s ...
```
Consider:

When session capability objects are omitted from functions generating session signatures, the SDK defaults to creating a session capability object with maximum permissions (i.e. can use any Lit Resource with any Lit Ability that the signer of the `AuthSig` owns). This should only be done when debugging, as allowing unspecified access control conditions is a security vulnerability.

Read more [here](https://deploy-preview-311--lit-dev-docs.netlify.app/sdk/authentication/session-sigs/capability-objects) about how to create custom session capability objects.

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
```diff
@@ -3,112 +3,153 @@ sidebar_position: 2
+import TabItem from '@theme/TabItem';
 
-# Generating SessionSigs
+# Generating SessionSigs: `getSessionSigs`
```
```suggestion
# Generating SessionSigs Using an Auth Sig
```

awisniew207 replied:
Did this, but `AuthSig` or `Auth Sig`?

awisniew207 replied:
Will make this Auth Sig, I think people will understand

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
```diff
@@ -3,112 +3,153 @@ sidebar_position: 2
-  * then the user will be prompted to authenticate with their wallet.
-  */
-  const authNeededCallback = async ({
+## Parameters and Returns
```
```suggestion
## Parameters and Returns Values
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
```diff
@@ -3,112 +3,153 @@ sidebar_position: 2
+</TabItem>
+</Tabs>
+
+The `node-localstorage` dependency is only required when executing code outside a browser environment. The SDK will use the native browser storage when in a browser environment.
```
```suggestion
The `node-localstorage` dependency is only required when executing code outside a browser environment. The SDK will use the native browser storage when in a browser environment. You can learn more about this [here](../intro#storing-sessionsigs).
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-session-sigs.md:
```diff
@@ -3,112 +3,153 @@ sidebar_position: 2
+
+To get the Lit resource identifier for other resources, you can use the other methods included in [@lit-protocol/auth-helpers](https://v6-api-doc-lit-js-sdk.vercel.app/modules/auth_helpers_src.html) package.
+
+If you would like to use this function on the `datil` or `datil-test` networks, a `capacityDelegationAuthSig` is required. An example of how to generate one can be found in the full code example.
```
Once #319 is merged, we should like to the Delegating a Credit page

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-pkp-session-sigs.md:
```diff
@@ -0,0 +1,148 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Generating SessionSigs: `getPkpSessionSigs`
```
```suggestion
# Generating SessionSigs Using a PKP
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-pkp-session-sigs.md:
```diff
@@ -0,0 +1,148 @@
+- [Lit Resources and Abilities](./resources-and-abilities.md)
+- [PKPs](../../wallets/minting)
+
+## Parameters and Returns
```
```suggestion
## Parameters and Return Values
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-pkp-session-sigs.md:
```diff
@@ -0,0 +1,148 @@
+
+## Parameters and Returns
+
+To see the parameters and return of `getPkpSessionSigs`, please visit our [API Docs](https://v6-api-doc-lit-js-sdk.vercel.app/classes/lit_node_client_src.LitNodeClientNodeJs.html#getPkpSessionSigs).
```
```suggestion
To see the parameters and return values of `getPkpSessionSigs`, please visit our [API Docs](https://v6-api-doc-lit-js-sdk.vercel.app/classes/lit_node_client_src.LitNodeClientNodeJs.html#getPkpSessionSigs).
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-pkp-session-sigs.md:
```diff
@@ -0,0 +1,148 @@
+</TabItem>
+</Tabs>
+
+The `node-localstorage` dependency is only required when executing code outside a browser environment. The SDK will use the native browser storage when in a browser environment.
```
```suggestion
The `node-localstorage` dependency is only required when executing code outside a browser environment. The SDK will use the native browser storage when in a browser environment. You can learn more about this [here](../intro#storing-sessionsigs).
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-lit-action-session-sigs.md:
```diff
@@ -0,0 +1,168 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Generating SessionSigs: `getLitActionSessionSigs`
```
```suggestion
# Generating SessionSigs Using a Lit Action
```

spacesailor24 commented on docs/sdk/authentication/session-sigs/get-lit-action-session-sigs.md:
```diff
@@ -0,0 +1,168 @@
+
+## Parameters and Returns
+
+To see the parameters and return of `getLitActionSessionSigs`, please visit our [API Docs](https://v6-api-doc-lit-js-sdk.vercel.app/classes/lit_node_client_src.LitNodeClientNodeJs.html#getLitActionSessionSigs).
```
```suggestion
To see the parameters and return values of `getLitActionSessionSigs`, please visit our [API Docs](https://v6-api-doc-lit-js-sdk.vercel.app/classes/lit_node_client_src.LitNodeClientNodeJs.html#getLitActionSessionSigs).
```

spacesailor24 reviewed (approved):
Awesome job 🙂
